internal/adapters/terraform/aws/autoscaling/adapt.go

```diff
@@ -0,0 +1,143 @@
+	var AvaiabilityZone []autoscaling.AutoscalingGroupsList
+	for _, module := range modules {
+		for _, resource := range module.GetResourcesByType("aws_autoscaling_group") {
+			AvaiabilityZone = append(AvaiabilityZone, adaptAutoscaling(resource, module))
+		}
+	}
+	return AvaiabilityZone
+}
+
+func adaptNotificationConfigurations(modules terraform.Modules) []autoscaling.NotificationConfigurations {
+	var NotificationConfig []autoscaling.NotificationConfigurations
+	for _, module := range modules {
+		for _, resource := range module.GetResourcesByType("aws_autoscaling_notification") {
+			NotificationConfig = append(NotificationConfig, adaptNotificationConfiguration(resource, module))
+		}
+	}
+	return NotificationConfig
+}
+
+func adaptLaunchConfigurations(modules terraform.Modules) []autoscaling.LaunchConfigurations {
+	var LaunchConfig []autoscaling.LaunchConfigurations
+	for _, module := range modules {
+		for _, resource := range module.GetResourcesByType("aws_launch_configuration") {
+			LaunchConfig = append(LaunchConfig, adaptLaunchConfiguration(resource, module))
+		}
+	}
+	return LaunchConfig
+}
+
+func adaptAutoscaling(resource *terraform.Block, module *terraform.Module) autoscaling.AutoscalingGroupsList {
+	nameAttr := resource.GetAttribute("name")
+	nameVal := nameAttr.AsStringValueOrDefault("", resource)
+
+	var AZones []types.StringValue
+	AZAttr := resource.GetAttribute("availability_zones")
+	for _, AZ := range AZAttr.AsStringValues() {
+		AZones = append(AZones, AZ)
+	}
+
+	HCTAttr := resource.GetAttribute("health_check_type")
+	HCTVal := HCTAttr.AsStringValueOrDefault("", resource)
+
+	var LBNames []types.StringValue
+	LBNAttr := resource.GetAttribute("load_balancers")
+	for _, LBN := range LBNAttr.AsStringValues() {
+		LBNames = append(LBNames, LBN)
+	}
+
+	ASGArnAttr := resource.GetAttribute("arn")
+	ASGArnVal := ASGArnAttr.AsStringValueOrDefault("", resource)
+
+	DefaultCooldownAttr := resource.GetAttribute("default_cooldown")
+	DefaultCooldownVal := DefaultCooldownAttr.AsIntValueOrDefault(0, resource)
+
+	LaunchConfigurationNameAttr := resource.GetAttribute("launch_configuration")
+	LaunchConfigurationNameVal := LaunchConfigurationNameAttr.AsStringValueOrDefault("", resource)
+
+	var suspendedprocess []autoscaling.SuspendedProcesses
+	for _, susBlock := range resource.GetBlocks("suspended_processes") {
+
+		suspendedprocess = append(suspendedprocess, autoscaling.SuspendedProcesses{
+			Metadata: susBlock.GetMetadata(),
+		})
+	}
+
+	var Tags []autoscaling.Tags
+	tagsRes := resource.GetBlocks("tags")
+	for _, tagRes := range tagsRes {
+
+		Tags = append(Tags, autoscaling.Tags{
+			Metadata:   tagRes.GetMetadata(),
+			ResourceId: types.StringDefault("", tagRes.GetMetadata()),
+		})
+	}
+
+	return autoscaling.AutoscalingGroupsList{
+		Metadata:                resource.GetMetadata(),
+		Name:                    nameVal,
+		AvaiabilityZone:         AZones,
+		Instances:               nil,
```

Comment:
ditto as well for instances being nil

Reply:
I used to place nil where I was unables to find values. Here in **instances** adapter need instances id and and metadata. but there were no instances id in the documentation. but I change that function with unresolveable strings and metadata? is this the way?

Reply:
Can you share a link to the documentation?

Reply:
Yes, Here is the [https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group](link)